### ISSUE TITLE
[Bug]: `sample_hypersphere` in 1D does not take random seed into account

### ISSUE THREAD

**Issue by wgsb** [pytorch/botorch#2685]
### What happened?

`sample_hypersphere` does not take random seed into account when `d=1`.

This can be simply fixed by changing `botorch/utils/sampling.py:165` from

```python
    if d == 1:
        rnd = torch.randint(0, 2, (n, 1), device=device, dtype=dtype)
        return 2 * rnd - 1
```

to

```python
    if d == 1:
        with manual_seed(seed):
            rnd = torch.randint(0, 2, (n, 1), device=device, dtype=dtype)
        return 2 * rnd - 1
```

### Please provide a minimal, reproducible example of the unexpected behavior.

```python
>>> from botorch.utils.sampling import sample_hypersphere
>>> def compare_samples(d, n=1_000, seed=0):
...     a = sample_hypersphere(d=d, n=n, seed=seed)
...     b = sample_hypersphere(d=d, n=n, seed=seed)
...     return a.equal(b)
... 
>>> [compare_samples(d) for d in range(1, 10)]
[False, True, True, True, True, True, True, True, True]

```

### Please paste any relevant traceback/logs produced by the example provided.

```shell

```

### BoTorch Version

0.12.0

### Code of Conduct

- [x] I agree to follow BoTorch's Code of Conduct

**Comment by saitcakmak** [comment]
Hi @wgsb. Thanks for flagging the issue! I'll put up a fix shortly
